packages/core/src/components/field-wrapper/helper-text-util.tsx

```diff
@@ -51,51 +57,52 @@ export function renderHelperText({
   validText?: string;
   helperText?: string;
 }) {
-  if (isInvalid && invalidText !== undefined) {
+  if (isInvalid && invalidText !== undefined && invalidText?.trim() !== '') {
```

Comment:
Please remove all redundant undefined checks
```suggestion
  if (isInvalid && invalidText?.trim() !== '') {
```